sample-3.md

```diff
@@ -0,0 +1,536 @@
+
+```shell
+kubectl -n $K_NAMESPACE get broker default
+```
+
+This should return something like:
+
+```shell
+NAME      READY   REASON   HOSTNAME                                                           AGE
+default   True             default-broker.kn-eventing-step-by-step-sample.svc.cluster.local   1m
+```
+
+We are particularly interested in the second column, `READY`. We want that value to
+be `True`. It may take a minute or two to become `True`. If it is not `True` within
+two minutes, then something has gone wrong. Hopefully the `REASON` column gives a
+useful error message. See our [Broker Debugging Guide](somewhere) to help fix the
+error.
+
+## Event Consumers
+
+The first thing we are going to do is create our event consumers. These are 
+`Deployment`s, and `Service`s that point at them. We are creating two distinct
+event consumers, so that we can show how to selectively send events to distinct
+event consumers later.
+
+Setup the '-dev' `Service` and `Deployment`.
+
+```shell
+cat << END | kubectl -n $K_NAMESPACE apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: event-display-dev
+spec:
+  replicas: 1
+  selector:
+    matchLabels: &labels
+      app: event-display-dev
+  template:
+    metadata:
+      labels: *labels
+    spec:
+      containers:
+        - name: event-display
+          # Source code: https://github.com/knative/eventing-sources/blob/release-0.6/cmd/event_display/main.go
+          image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display@sha256:37ace92b63fc516ad4c8331b6b3b2d84e4ab2d8ba898e387c0b6f68f0e3081c4
+
+---
+
+# Service pointing at the previous Deployment. This will be the target for event
+# consumption.
+kind: Service
+apiVersion: v1
+metadata:
+  name: event-display-dev
+spec:
+  selector:
+    app: event-display-dev
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 8080
+END
+```
+
+Now setup the '-test' `Service` and `Deployment`:
+
+```shell
+cat << END | kubectl -n $K_NAMESPACE apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: event-display-test
+spec:
+  replicas: 1
+  selector:
+    matchLabels: &labels
+      app: event-display-test
+  template:
+    metadata:
+      labels: *labels
+    spec:
+      containers:
+        - name: event-display
+          # Source code: https://github.com/knative/eventing-sources/blob/release-0.6/cmd/event_display/main.go
+          image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display@sha256:37ace92b63fc516ad4c8331b6b3b2d84e4ab2d8ba898e387c0b6f68f0e3081c4
+
+---
+
+# Service pointing at the previous Deployment. This will be the target for event
+# consumption.
+kind: Service
+apiVersion: v1
+metadata:
+  name: event-display-test
+spec:
+  selector:
+    app: event-display-test
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 8080
+END
+```
+
+Let's wait for the `Deployment`s to be ready, as shown by their Available status
```

Comment:
'Deployments'? current markdown creates space between t and s.

Reply:
I have done this all over. The idea being to match the K8s resource name. Should I change all of them?